sdk/core/azure-core-experimental/src/main/java/com/azure/core/experimental/util/BinaryData.java

```diff
@@ -18,18 +20,26 @@
 import java.io.InputStream;
 import java.io.UncheckedIOException;
 import java.nio.ByteBuffer;
-import java.nio.charset.Charset;
 import java.nio.charset.StandardCharsets;
 import java.util.Arrays;
 import java.util.Objects;
 
 /**
- * This class is an abstraction over many different ways that binary data can be represented. The {@link BinaryData}
- * can be created from {@link InputStream}, {@link Flux} of {@link ByteBuffer}, {@link String}, {@link Object} and byte
- * array. The data is collected from provided sources and stored into a byte array.
+ * This class is an abstraction over many different ways that binary data can be represented. The data represented by
+ * {@link BinaryData} is immutable. The {@link BinaryData} can be created from {@link InputStream}, {@link Flux} of
```

Comment:
"The data represented by {@link BinaryData} is immutable" is an odd statement to make? It is taken into BinaryData and presumably copied, but when given back to the user is mutable again. Perhaps clarify what you mean by this statement.